### PR TITLE
[AMP OP&Test]Add fp16/bf16 support isnan/isfinite/isinf op

### PR DESCRIPTION
**Pull request by wjj19950828**
### PR types

Performance optimization
### PR changes

OPs
### Describe

- Add fp16/bf16 support isnan/isfinite/isinf op

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_isfinite_op.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 import numpy as np
 from eager_op_test import OpTest
+from op_test import convert_float_to_uint16
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_isfinite_op.py]: 从`eager_op_test`中import

Comment (wjj19950828) [reply python/paddle/fluid/tests/unittests/test_isfinite_op.py]: Done.

File: python/paddle/fluid/tests/unittests/test_isfinite_op.py [hunk 2]
```diff
@@ -48,6 +48,26 @@ def init_dtype(self):
...
+        x = np.random.uniform(0.1, 1, [11, 17]).astype(np.float32)
+        x[0] = np.inf
+        x[-1] = np.inf
+
+        out = np.isinf(x).astype(np.float32)
+        self.inputs = {'X': convert_float_to_uint16(x)}
+        self.outputs = {'Out': out}
+
+    def test_output(self):
+        self.check_output()
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_isfinite_op.py]: 这里需要直接调用check_output_with_place，不然CPU的place会报错

File: python/paddle/fluid/tests/unittests/test_isfinite_op.py [hunk 3]
```diff
@@ -48,6 +48,26 @@ def init_dtype(self):
         self.dtype = np.float16
 
 
+# BFP16 isinf Test
+@unittest.skipIf(
+    not core.is_compiled_with_cuda(), "core is not compiled with CUDA"
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_isfinite_op.py]: 这里也需要加一下gpu是否支持bf16的判断

File: python/paddle/fluid/tests/unittests/test_isfinite_op.py (hunk 2 above)
Comment (wjj19950828) [reply python/paddle/fluid/tests/unittests/test_isfinite_op.py]: Done.

File: python/paddle/fluid/tests/unittests/test_isfinite_op.py (hunk 3 above)
Comment (wjj19950828) [reply python/paddle/fluid/tests/unittests/test_isfinite_op.py]: Done.

**Review by ZzSean — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]

**Pull request merged**